### ISSUE TITLE
updating scalafmtConfig does not invalidate the scalafmt/scalafmtCheck caches

### ISSUE THREAD

**Issue by bjaglin** [scalameta/sbt-scalafmt#41]
Actual (cold start, 58 sources already formatted)
```
$ scalafmt
[info] Formatting 58 Scala sources...
[success] Total time: 3 s, completed Aug 5, 2019, 6:32:11 PM
$ set scalafmtConfig := file(".scalafmt2.conf")
[info] Reapplying settings...
$ scalafmt
[success] Total time: 0 s, completed Aug 5, 2019, 6:32:51 PM
```

Expected (cold start, 58 sources already formatted)
```
$ scalafmt
[info] Formatting 58 Scala sources...
[success] Total time: 3 s, completed Aug 5, 2019, 6:32:11 PM
$ set scalafmtConfig := file(".scalafmt2.conf")
[info] Reapplying settings...
$ scalafmt
[info] Formatting 58 Scala sources...
[success] Total time: 3 s, completed Aug 5, 2019, 6:32:51 PM
```

**Comment by poslegm** [comment]
Related discussion https://github.com/scalameta/scalafmt/issues/1401

**Comment by bjaglin** [comment]
Note that there is another level of caching in `ScalafmtDynamic`, which is broken when fed with an invalid config file: https://github.com/scalameta/scalafmt/issues/1442

**Comment by bjaglin** [comment]
Fixed by #40 (automatic close didn't work as issue was not referenced properly)
